### ISSUE TITLE
Remove tools-local from repo root

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/runtime#36519]
As I was just working in tools-local on the tasks composition I was wondering if the "tools-local" folder needs to live under the repo root. I propose to move its contents into the "src" folder:

```
src
- libraries
- installer
- coreclr
- mono
- tasks (local msbuild tasks)
- tools (global and repo dotnet tools)
- scripts (we should move https://github.com/dotnet/runtime/blob/master/src/libraries/FindResourceDuplicates.ps1 and https://github.com/dotnet/runtime/blob/master/src/libraries/GenerateLibrariesSln.ps1 there as well)
```

cc @safern @danmosemsft

**Comment by jkotas** [comment]
> - tools (global and repo dotnet tools)

Are these tools just for the internal repo consumption (e.g. they will never ship or publish an artifact to nuget feed)? 

If it is just for tools consumed internally by the repo, I like "-local" in the current directory name. It makes it clear that it is not for any kind of tools.

If it is for any kind of tools, what is going to be the rule for when the tool should live here vs. somewhere else?

> scripts

We have majority of the scripts under `eng`, and one-off scripts live to their use today. 

What is going to be the rule for when the script should live here vs. somewhere else? What is the list of scripts that would move here?

**Comment by ViktorHofer** [comment]
> Are these tools just for the internal repo consumption (e.g. they will never ship or publish an artifact to nuget feed)?

Yes those tools should never ship and are intended to be used in the repository. We currently have one single tool which applies to that which is the [Debian Packaging Tool](https://github.com/dotnet/runtime/tree/master/tools-local/dotnet-deb-tool). I agree that "tools-local" then probably makes more sense.

> What is going to be the rule for when the script should live here vs. somewhere else? What is the list of scripts that would move here?

Primarily the scripts that currently already live under tools-local: https://github.com/dotnet/runtime/tree/master/tools-local/scripts. Maybe also https://github.com/dotnet/runtime/blob/master/src/libraries/FindResourceDuplicates.ps1, https://github.com/dotnet/runtime/blob/master/src/libraries/GenerateLibrariesSln.ps1 if they are useful outside of libraries.

Another suggestion would be to just move the "tools-local" folder as-is into the src folder.

**Comment by jkotas** [comment]
> Debian Packaging Tool

This looks like one-off tool for the installer. Should it better live somewhere under src/installer?

We have number of other similar one-off tools. For example, https://github.com/dotnet/runtime/tree/master/src/coreclr/src/tools/Common/JitInterface/ThunkGenerator is a tool to generate jit/ee interface definitions.

> Primarily the scripts that currently already live under tools-local

Why not move the script that currently live under tools-local to be under eng where majority of the scripts live? I do not see a clear difference between the nature of the scripts under tools-local vs eng.

**Comment by ViktorHofer** [comment]
> This looks like one-off tool for the installer. Should it better live somewhere under src/installer?

I opened https://github.com/dotnet/runtime/issues/36537 to discuss the future of that tool and where it should live.

> Why not move the script that currently live under tools-local to be under eng where majority of the scripts live? I do not see a clear difference between the nature of the scripts under tools-local vs eng.

Would work for me as well if those are scripts which aren't supposed to be invoked directly by us devs.

**Comment by jkotas** [comment]
I have looked at the scripts under https://github.com/dotnet/runtime/tree/master/tools-local/scripts and all of them look dead / no longer used. Maybe we should just delete it and see whether we will miss any of them.

That just leaves the tasks. We have been aggressively moving all tasks to arcade, even if they are not really useful outside dotnet/runtime. Would it make sense to make the installer task to arcade as well?

**Comment by ViktorHofer** [comment]
> I have looked at the scripts under https://github.com/dotnet/runtime/tree/master/tools-local/scripts and all of them look dead / no longer used. Maybe we should just delete it and see whether we will miss any of them.

Thanks for doing that in https://github.com/dotnet/runtime/commit/56ee1b0b3f78d03af75434224ea9bd0cd54ec51f#diff-0b05377ea358dd4929229ced386ea133.

> That just leaves the tasks. We have been aggressively moving all tasks to arcade, even if they are not really useful outside dotnet/runtime. Would it make sense to make the installer task to arcade as well?

Some of these tasks are very specific to our repository, I don't think it makes sense to port them to Arcade. We already moved most non repo specific ones over.

**Comment by jkotas** [comment]
https://github.com/dotnet/runtime/blob/master/tools-local/tasks/installer.tasks have quite a bit of duplication with tasks in the installer repo. For example:

- https://github.com/dotnet/runtime/blob/master/tools-local/tasks/installer.tasks/GenerateMsiVersion.cs
- https://github.com/dotnet/installer/blob/master/src/core-sdk-tasks/GenerateMsiVersion.cs + https://github.com/dotnet/installer/blob/master/src/core-sdk-tasks/Utils/Version.cs
